### PR TITLE
Merge /agent/demo and /browser/demo into one root /demo folder

### PR DESCRIPTION
**Pull request by eric-waldman**

### CODE REVIEW

File: agent/include/content_analysis/sdk/analysis_agent.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include <memory>
 #include <string>
 
-#include "content_analysis/sdk/analysis_agent.pb.h"
+#include "content_analysis/sdk/analysis.pb.h"
```
Comment (rogerta) [inline agent/include/content_analysis/sdk/analysis_agent.h:11]: Let's fix this in previous PR to make sure all commits can build successfully.

File: browser/include/content_analysis/sdk/analysis_client.h [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #include <memory>
 #include <string>
 
-#include "content_analysis/sdk/analysis_agent.pb.h"
+#include "content_analysis/sdk/analysis.pb.h"
```
Comment (rogerta) [inline browser/include/content_analysis/sdk/analysis_client.h:11]: Let's fix this in previous PR to make sure all commits can build successfully.

File: demo/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 The Chromium Authors.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+cmake_minimum_required(VERSION 3.22)
+
+project(chrome_enterprise_connector_local_analysis)
+
+# Makes available the package definitions in vcpkg.
+include(build/vcpkg/scripts/buildsystems/vcpkg.cmake)
```
Comment (rogerta) [inline demo/CMakeLists.txt:9]: Why is this line needed?  It also seems to hard code `build/...` as the directory where the demo is built.  Can you try specifying a directory outside the source tree to make that works too?

If this line is needed I think you need to replace`build` with `${PROJECT_BINARY_DIR}`

File: demo/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+find_package(Protobuf CONFIG REQUIRED)
+
+# The include directory contains the header files needed by the demo code.
+# The gen directory contains generated protobuf headers describing the request
+# and response objects used to communicate with Google Chrome.
+include_directories(
+  "${PROJECT_SOURCE_DIR}/../agent/include"
+  "${PROJECT_SOURCE_DIR}/../browser/include"
+  "${PROJECT_BINARY_DIR}/gen"
+)
```
Comment (rogerta) [inline demo/CMakeLists.txt:20]: Let's not mix the agent include path with the client include path.  To specify different include paths for agent and client, use [target_include_directories](https://cmake.org/cmake/help/git-master/command/target_include_directories.html).

Maybe remove the `include_directories` command and define two variables:
```
set(AGENT_INCLUDE_PATH
  "${PROJECT_SOURCE_DIR}/../agent/include"
  "${PROJECT_BINARY_DIR}/gen"
)

set(CLIENT_INCLUDE_PATH
  "${PROJECT_SOURCE_DIR}/../browser/include"
  "${PROJECT_BINARY_DIR}/gen"
)
```
then add the appropriate `target_include_directories(<target> <TARGET>_INCLUDE_PATH)` lines under the corresponding `add_library()` and `add_executable()` commands.

File: demo/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  ../agent/src/session_base.cc
+  ../agent/src/session_base.h
+  ${PLATFORM_AGENT_CODE}
+  "${PROJECT_BINARY_DIR}/gen/content_analysis/sdk/analysis.pb.cc"
+)
+target_link_libraries(cac_agent PUBLIC protobuf::libprotoc protobuf::libprotobuf protobuf::libprotobuf-lite)
+
+# Builds the content analysis connector browser linker library.  This library is linked
+# into the client in order to send content analysis requests to the agent.
+add_library(cac_browser
```
Comment (rogerta) [inline demo/CMakeLists.txt:72]: Some targets use the word `browser` and others use the word `client`.  How about making them all consistent?

**Review by rogerta — COMMENTED** [review]
Thanks Eric.  See comments below.  Should the file `analysis.proto` be renamed in this PR or the previous one?

File: agent/include/content_analysis/sdk/analysis_agent.h (hunk 1 above)
Comment (eric-waldman) [reply agent/include/content_analysis/sdk/analysis_agent.h:11]: Done.

File: browser/include/content_analysis/sdk/analysis_client.h (hunk 2 above)
Comment (eric-waldman) [reply browser/include/content_analysis/sdk/analysis_client.h:11]: Done.

File: demo/CMakeLists.txt (hunk 5 above)
Comment (eric-waldman) [reply demo/CMakeLists.txt:72]: Done.

File: demo/CMakeLists.txt (hunk 4 above)
Comment (eric-waldman) [reply demo/CMakeLists.txt:20]: Done.

**Review by rogerta — COMMENTED** [review]
Looking good, see below.  I get an error when I try to compile on linux:

`fatal error: content_analysis/sdk/analysis_agent.h: No such file or directory`
